aredis_om/model/model.py

```diff
@@ -425,11 +447,26 @@ def query(self):
         if self._query:
             return self._query
         self._query = self.resolve_redisearch_query(self.expression)
+        if self.knn:
+            self._query = (
+                self._query
+                if self._query.startswith("(") or self._query == "*"
+                else f"({self._query})"
+            ) + f"=>[{self.knn}]"
         return self._query
 
+    @property
+    def query_params(self):
+        params: List[Union[str, bytes]] = []
+        if self.knn:
```

Comment:
Any reason to not:

```return [attr for kv in self.knn.query_params.items() for attr in kv]```

Worst case scenario you should be getting an empty list, which is first case

Reply:
I expect future contributions to add on to this logic, and this allows that without having to change any lines.

Reply:
```py
def query_params(self):
    return [attr for kv in self.knn.query_params.items() for attr in kv]
```
This would return an error when `self.knn` is not set and `query_params` is accessed.